### PR TITLE
Create Generative_Adversarial_Networks.md

### LINKED ISSUE #1117: Generative Adversarial Networks (GAN)
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md).

### Description

GANs are a class of unsupervised learning algorithms that consist of two neural networks: a generator and a discriminator. The generator creates new data samples, 
while the discriminator tries to distinguish between real and generated data. GANs are used for tasks like image generation, style transfer, and data augmentation.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by inkerton**
### Description
Generative Adversarial Networks (GANs) are a class of deep learning models designed to generate new data samples that are similar to a given dataset. GANs consist of two neural networks, the generator and the discriminator, which are trained simultaneously in a game-theoretic setting. The generator learns to produce realistic data, while the discriminator tries to distinguish between real and generated data. This adversarial process helps the generator improve over time.

Fixes #1117 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
uses LaTeX equations with `$`

**Comment by inkerton** [comment]
done

File: docs/machine-learning/Generative_Adversarial_Networks.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  A common issue in GANs where the generator produces limited or repetitive outputs, failing to capture the full diversity of the training data.
+
+- **Wasserstein GAN (WGAN)**:  
+  An improved GAN variant that uses the Wasserstein distance as a loss function, helping to stabilize training and reduce mode collapse.
+
+### Example of GAN Architecture:
+
+1. **Generator Network**:  
+   - Input: Random noise vector (e.g., \( z \sim N(0, 1) \)).
+   - Output: Synthetic data (e.g., an image).
```
Comment (ajay-dhangar) [inline docs/machine-learning/Generative_Adversarial_Networks.md]: use `$` for `\( z \sim N(0, 1) \))`

File: docs/machine-learning/Generative_Adversarial_Networks.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+   The generator is a neural network that takes random noise as input and generates synthetic data (e.g., images, text) that resemble the real data. The goal of the generator is to produce data that the discriminator cannot distinguish from real data.
+
+2. **Discriminator**:  
+   The discriminator is another neural network that takes both real data and generated data as input and tries to classify whether the input is real (from the dataset) or fake (generated by the generator). The discriminator's task is binary classification.
+
+3. **Adversarial Process**:  
+   The generator and discriminator play a two-player minimax game, where the generator tries to fool the discriminator, and the discriminator tries not to be fooled. The objective of the generator is to minimize the discriminator's accuracy, while the discriminator's goal is to maximize its classification accuracy.
+
+### Objective Function:
+The objective function for GANs is a **minimax game** where the generator \( G \) tries to minimize the loss, and the discriminator \( D \) tries to maximize it. This is given by:
```
Comment (ajay-dhangar) [inline docs/machine-learning/Generative_Adversarial_Networks.md]: use `$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code

File: docs/machine-learning/Generative_Adversarial_Networks.md (hunk 2 above)
Comment (inkerton) [reply docs/machine-learning/Generative_Adversarial_Networks.md]: Done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**